### DISCUSSION TITLE
Filter on Relation Field in Field Collection

### DISCUSSION THREAD

**Discussion by YaseeraCentric** [pimcore/pimcore#15727] [Q&A / Support] [answered]
i have a class in which there is a field collection and in that field collection i have a ManyToOne Relation Column , so i want to apply filter on this column , this is my code.
```
$loyaltyProgram = new DataObject\LoyaltyProgram\Listing();
$loyaltyProgram->setCondition("`TierPoints`.MinTransaction = ? AND `TierPoints`.Tier__id IN ?", [300,$transaction->getCustomer()->getTier()->getId()]);
```
But I'm getting this error:
Unknown Column TierPoints`.Tier__id 

@blankse any help?

**Comment by blankse** [comment]
@YaseeraCentric You need to add the field collection to the listing. So the query have a join to the db table.
```
$loyaltyProgram->addFieldCollection('TierPoints', 'fieldCollectionFieldName');
```

**Reply by YaseeraCentric** [reply]
i did. this is my complete code
```
$loyaltyProgram = new DataObject\LoyaltyProgram\Listing();
             $loyaltyProgram->addConditionParam("StoreID REGEXP CONCAT('(^|,)(', REPLACE('$store_id',',', '|'), ')(,|$)')");
            
          
            $loyaltyProgram->addFieldCollection("TierPoints");
           $loyaltyProgram->setCondition("`TierPoints`.MinTransaction = ? AND `TierPoints`.Tier__id IN ?", [300,$transaction->getCustomer()->getTier()->getId()]);
            $loyaltyProgram->load();
```
@blankse

**Reply by YaseeraCentric** [reply]
@blankse any help?

**Reply by blankse** [reply]
@YaseeraCentric Can you show the complete error message with the complete sql query? Can you look in the db tables? Maybe the column have a different name?

**Reply by blankse** [reply]
@YaseeraCentric And why do you use the `IN` operator here? I think it should be a `=`.

**Reply by YaseeraCentric** [reply]
this is the complete error message
**An exception occurred while executing 'SELECT object_loyaltyprog.o_id as o_id, object_loyaltyprog.o_type as o_type FROM object_loyaltyprog LEFT JOIN object_collection_TierPoints_loyaltyprog `TierPoints` ON 1
AND `TierPoints`.o_id = `object_loyaltyprog`.o_id WHERE (`TierPoints`.MinTransaction = ? AND `TierPoints`.Tier__id IN (?) AND (StoreID REGEXP CONCAT('(^|,)(', REPLACE('47',',', '|'), ')(,|$)')) AND object_loyaltyprog.o_type IN ('object','folder')) AND object_loyaltyprog.o_published = 1' with params [300, 49]:**

SQLSTATE[42S22]: Column not found: 1054 Unknown column 'TierPoints.Tier__id' in 'where clause'
i checked the db tables the column has the same name i used in the query

**Reply by YaseeraCentric** [reply]
**@YaseeraCentric And why do you use the IN operator here? I think it should be a =.**
REPLY:
i tried with both IN and = but getting the same error i mentioned
@blankse

**Reply by blankse** [reply]
Which db columns have the `object_collection_TierPoints_loyaltyprog` db table? You can look with "Database Admin" menu item in the pimcore backend.

**Reply by YaseeraCentric** [reply]
this is the Loyalty Program Relation tables structure
![image](https://github.com/pimcore/pimcore/assets/125441244/60ecf9bf-c888-44ac-8979-6eee16dd9169)

**Reply by YaseeraCentric** [reply]
@blankse  this is the object_collection_TierPoints_loyaltyprog table
![image](https://github.com/pimcore/pimcore/assets/125441244/bbe6ab61-c479-46cd-b450-aa6b3f617048)

**Reply by YaseeraCentric** [reply]
@blankse this is the pimcore class structure
![image](https://github.com/pimcore/pimcore/assets/125441244/24b718ba-1b8f-459a-8f7d-e94fb6b4b31f)

**Reply by YaseeraCentric** [reply]
@blankse and this is the field collection "TierPoints"
![image](https://github.com/pimcore/pimcore/assets/125441244/e5b57b7a-02d7-4ce9-9a4d-98618ba56630)

**Reply by YaseeraCentric** [reply]
@blankse any help?

**Comment by blankse** [comment]
@YaseeraCentric Try this:
```
$loyaltyProgram->setCondition("`TierPoints`.MinTransaction = ? AND `TierPoints`.o_id IN (SELECT src_id FROM object_relations_loyaltyprog WHERE dest_id = ? AND type = 'object' AND fieldname = 'Tier' AND ownertype = 'fieldcollection' AND ownername = 'TierPoints')", [300, $transaction->getCustomer()->getTier()->getId()]);
```

**Reply by YaseeraCentric** [reply]
@blankse and how can i access it after , like that?
 foreach($loyaltyProgram as $program){
                dd($program->getTierPoints()->getTier());
            }

**Reply by blankse** [reply]
```
foreach($loyaltyProgram->getData() as $program){
    dd($program->getTierPoints()->getTier());
}
```

**Reply by YaseeraCentric** [reply]
@blankse  getting this error
Call to undefined method getTier in class Pimcore\Model\DataObject\Fieldcollection

**Reply by blankse** [reply, marked as answer]
```
foreach ($loyaltyProgram->getData() as $program){
    foreach ($program->getTierPoints()->getItems() as $tierPoint) {
       dd($tierPoint->getTier());
    }
}
```
